rest-api/src/main/java/gov/cms/qpp/conversion/api/controllers/v1/CpcFileControllerV1.java

```diff
@@ -26,21 +35,50 @@
 	@Autowired
 	private CpcFileService cpcFileService;
 
+	@Autowired
+	private FileRetrievalService fileRetrievalService;
+
 	/**
 	 * Endpoint to transform an uploaded file into a valid or error json response
 	 *
 	 * @return Valid json or error json content
-	 * @throws IOException If errors occur during file upload or conversion
 	 */
 	@RequestMapping(method = RequestMethod.GET, value = "/unprocessed-files",
-			headers = {"Accept=" + Constants.V1_API_ACCEPT + ", application/json;charset=UTF-8"})
+			headers = {"Accept=" + Constants.V1_API_ACCEPT})
 	public ResponseEntity<List> getUnprocessedCpcPlusFiles() throws IOException {
 		API_LOG.info("CPC+ unprocessed files request received");
 
 		List<UnprocessedCpcFileData> unprocessedCpcFileDataList = cpcFileService.getUnprocessedCpcPlusFiles();
 
 		API_LOG.info("CPC+ unprocessed files request succeeded");
 
-		return new ResponseEntity<>(unprocessedCpcFileDataList, HttpStatus.OK);
+		HttpHeaders httpHeaders = new HttpHeaders();
+		httpHeaders.setContentType(MediaType.APPLICATION_JSON_UTF8);
+
+		return new ResponseEntity<>(unprocessedCpcFileDataList, httpHeaders, HttpStatus.OK);
+	}
+
+	/**
+	 * Retrieve a stored S3 object.
+	 *
+	 * @param fileId id for the stored object
+	 * @return object json or xml content
+	 * @throws IOException if S3Object content stream is invalid
+	 */
+	@RequestMapping(method = RequestMethod.GET, value = "/get-file/{fileId}",
+			headers = {"Accept=" + Constants.V1_API_ACCEPT})
+	public ResponseEntity<String> getFileById(@PathVariable("fileId") String fileId)
+			throws IOException {
+		API_LOG.info("CPC+ file request received");
+
+		CompletableFuture<InputStream> fileStreamFuture = fileRetrievalService.getFileById(fileId);
+		InputStream inStream = fileStreamFuture.join();
+
+		API_LOG.info("CPC+ file request succeeded");
+
+		HttpHeaders httpHeaders = new HttpHeaders();
+		httpHeaders.setContentType(MediaType.APPLICATION_XML);
+
+		return new ResponseEntity<>(IOUtils.toString(inStream, Charset.defaultCharset()), httpHeaders, HttpStatus.ACCEPTED);
```

Comment:
Why a status of ACCEPTED here rather than OK?  This will be returned synchronously.

Reply:
👍